app/controllers/users_controller.rb

```diff
@@ -188,4 +188,18 @@ def handle_settings_tab
       not_found unless @tab_list.map { |t| t.downcase.tr(" ", "-") }.include? @tab
     end
   end
+
+  # This endpoint is only for development as
+  # a way to quicklying login
+  def quickin
```

Comment:
I think it's helpful for development. This should be wrapped in `Rails.env.development?` though, like so:
```
class UsersController < ApplicationController
  # other methods...

  if Rails.env.development?
    def quickin
      # etc etc
    end
  end
# rest of controller methods
end
```

Reply:
@rhymes I added this feature as it was suggested by @Zhao-Andy from another ticket for me to put forth this functionality as it would be a good addition to help others new to the codebase to quickly impersonate users.

I do have a fail safe to ensure that this only works with development and return a 404 in anything but developement.
 
```ruby
  def quickin
    return raise ActionController::RoutingError, "Not Found" unless Rails.env.development?
  end
```

@Zhao-Andy you're suggested alternative syntax interests as me as its appears more explicit. I am uncertain if an if statement is allowed to be used in this manner but I give it a try as I am quite curious.

```
  if Rails.env.development?
    def quickin
      # etc etc
    end
  end
```

@rhymes There are two places were we can prevent this being used in production, here as we already do in this function and as @Zhao-Andy suggested in the routes.

Reply:
@omenking got it. I just think it's a dangerous endpoint to have if for any reason the `if Rails.env.development?` disappears in the future, one will be able impersonate anybody on the website. I'd figure out a way to have at least a test that fails if that happens :)

As to the "guard code" it could be with the 404 error, with the if before as the second example or in the `routes.rb` file, I don't know which is best.